src/components/Network.vue

```diff
@@ -0,0 +1,216 @@
+                keyguardRequest.senderType || Nimiq.Account.Type.BASIC,
+                new Nimiq.Address(new Nimiq.SerialBuffer(keyguardRequest.recipient)),
+                keyguardRequest.recipientType || Nimiq.Account.Type.BASIC,
+                keyguardRequest.value,
+                keyguardRequest.fee,
+                keyguardRequest.validityStartHeight,
+                keyguardRequest.flags || 0,
+                new Nimiq.SerialBuffer(keyguardRequest.data || 0),
+                Nimiq.SignatureProof.singleSig(
+                    new Nimiq.PublicKey(keyguardResult.publicKey),
+                    new Nimiq.Signature(keyguardResult.signature),
+                ).serialize(),
+                keyguardRequest.networkId,
+            );
+        } else {
+            tx = new Nimiq.BasicTransaction(
+                new Nimiq.PublicKey(keyguardResult.publicKey),
+                new Nimiq.Address(new Nimiq.SerialBuffer(keyguardRequest.recipient)),
+                keyguardRequest.value,
+                keyguardRequest.fee,
+                keyguardRequest.validityStartHeight,
+                new Nimiq.Signature(keyguardResult.signature),
+                keyguardRequest.networkId,
+            );
+        }
+
+        return tx;
+    }
+
+    /**
+     * The result of this method can also be used as a PlainTransaction
+     * for all relevant nano-api methods.
+     */
+    public makeSignTransactionResult(tx: Nimiq.Transaction): SignTransactionResult {
+        const proof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
+
+        const result: SignTransactionResult = {
+            serializedTx: tx.serialize(),
+
+            sender: tx.sender.toUserFriendlyAddress(),
+            senderType: tx.senderType,
+            senderPubKey: proof.publicKey.serialize(),
+
+            recipient: tx.recipient.toUserFriendlyAddress(),
+            recipientType: tx.recipientType,
+
+            value: Nimiq.Policy.satoshisToCoins(tx.value),
+            fee: Nimiq.Policy.satoshisToCoins(tx.fee),
+            validityStartHeight: tx.validityStartHeight,
+
+            signature: proof.signature.serialize(),
+
+            extraData: tx.data,
+            flags: tx.flags,
+            networkId: tx.networkId,
+
+            hash: tx.hash().toBase64(),
+        };
+
+        return result;
+    }
+
+    /**
+     * Relays the transaction to the network and only returns when the network
+     * fires its 'transaction-relayed' event for that transaction.
+     */
+    public async sendToNetwork(tx: Nimiq.Transaction): Promise<SignTransactionResult> {
+        const txObj = this.makeSignTransactionResult(tx);
+        const client = await this._getNetworkClient();
+
+        client.relayTransaction(txObj);
+
+        return new Promise<SignTransactionResult>((resolve, reject) => {
+            this.$once('transaction-relayed', (txInfo: any) => {
+                if (txInfo.hash === txObj.hash) resolve(txObj);
```

Comment:
It is guaranteed in that if the transaction is really relayed, the event is guaranteed to fire. But you are right, this needs a timeout.

Reply:
so as you said in the other conversation, this does in fact not block until the event fires, which the comment above it suggests. In case that it really only should return once (if) the event fires this seems not sufficient.